src/main/scala/com/wavesplatform/serialization/Deser.scala

```diff
@@ -6,7 +6,10 @@ object Deser {
 
   def serializeBoolean(b: Boolean): Array[Byte] = if (b) Array(1: Byte) else Array(0: Byte)
 
-  def serializeArray(b: Array[Byte]): Array[Byte] = Shorts.toByteArray(b.length.toShort) ++ b
+  def serializeArray(b: Array[Byte]): Array[Byte] = {
+    val lengthBytes = Shorts.toByteArray(b.length.toShort)
```

Comment:
There should be an overflow check
```suggestion
    val lengthBytes = Shorts.toByteArray(b.length.ensuring(_.isValidShort).toShort)
```

Reply:
fixed